src/DataManagement/Goals/GoalManager.cs

```diff
@@ -19,59 +22,116 @@ public GoalManager(IGoalList goalList, int userId = 0)
 
         public bool Add(IGoal goal)
         {
-            var id = GoalDbUpdater.Add(goal, UserId);
+            var id = GoalDatabase.Create(goal, UserId);
             //Check if id is correct, return false if something is wrong
-
             goal.Id = id;
-            goal.UserId = UserId;
             GoalList.Add(goal);
-            return false;
+            return true;
```

Comment:
This change is interesting, why false removed and changed to true?

Reply:
This was a mistake from before.
In general the Data Managers and Database classes lack proper exception handling. So as you see, some of them are returning true always, but I want to make them return false when something went wrong